[PyROOT] Add test for using function decl

stwunsch opened the pull request:
Reported on mattermost that "import ROOT; ROOT.TGraphMultiErrors().SetLineColor(0)"
breaks with 6.22 and is now fixed in master due to the ROOT PR #6277.

stwunsch commented:
@etejedor Is this the correct place for the test?

etejedor reviewed (changes requested):
Thanks @stwunsch ! I wonder if we could create a new class of tests for using declarations here:

https://github.com/root-project/roottest/blob/master/python/cpp/PyROOT_cpptests.py

since this is likely not the only test we will want to add for using declarations (there are several tickets related to it).

stwunsch commented:
> Thanks @stwunsch ! I wonder if we could create a new class of tests for using declarations here:
> 
> https://github.com/root-project/roottest/blob/master/python/cpp/PyROOT_cpptests.py
> 
> since this is likely not the only test we will want to add for using declarations (there are several tickets related to it).

I've moved the test and introduced a new class of tests related to using declarations!
